lib/internal/blob.js

```diff
@@ -56,17 +60,41 @@ const {
 const kHandle = Symbol('kHandle');
 const kType = Symbol('kType');
 const kLength = Symbol('kLength');
+const kArrayBufferPromise = Symbol('kArrayBufferPromise');
 
 const disallowedTypeCharacters = /[^\u{0020}-\u{007E}]/u;
 
 let Buffer;
+let ReadableStream;
+let threadId;
+let URL;
+
+function lazyURL(id) {
+  if (URL === undefined)
+    URL = require('url').URL;
+  return new URL(id);
+}
 
 function lazyBuffer() {
   if (Buffer === undefined)
     Buffer = require('buffer').Buffer;
   return Buffer;
 }
 
+function lazyReadableStream(options) {
+  if (ReadableStream === undefined) {
+    ReadableStream =
+      require('internal/webstreams/readablestream').ReadableStream;
+  }
+  return new ReadableStream(options);
+}
+
+function lazyThreadId() {
+  if (threadId === undefined)
+    threadId = require('worker_threads').threadId;
+  return threadId;
```

Comment:
nit
```suggestion
  return threadId ??= require('worker_threads').threadId;
```

Reply:
You could just grab `threadId` from `internalBinding('worker')` and skip the lazy-loading step here.

(Aside: Don’t we have snapshot support for a lot of builtins now? Isn’t lazy loading actively making things slower?)

Reply:
I'm removing the thread ID. Unfortunately the lazy loading is required because of circular dependencies between `buffer`, `blob`, and `url`.